### DISCUSSION TITLE
OpenAI API Key requirement

### DISCUSSION THREAD

**Discussion by eastbowl** [karthink/gptel#169] [General]
Hi,

This is probably a very naive question (I'm a novice and non-programmer).  I'm trying to use your module/package with a llama server on my computer.   When invoking "M-x gptel" I am asked for my OpenAI API key.  (My API key has no credits, so it fails to permit any access). 

My questions is, Why is it necessary to even contact OpenAI when using a completely different LLM on my own computer?  I'm sure you have a good reason for it.  I'm just curious what that reason is.  It would help me understand things (possibly).  

Thanks for your time.

**Comment by karthink** [comment]
>  When invoking "M-x gptel" I am asked for my OpenAI API key.

What is the default value of `gptel-backend`? With your cursor in the buffer from which you called `gptel`, could you run `M-x eval-expression` `(gptel-backend-name gptel-backend)`?

**Reply by eastbowl** [reply]
When I eval the variable gptel-backend in the buffer in which i call gptel, I get the following:

#s(gptel-openai "ChatGPT" "api.openai.com" #[0 "��� PBC�" ["Authorization" "Bearer " gptel--get-api-key] 3] "https" t "/v1/chat/completions" gptel-api-key ("gpt-3.5-turbo" "gpt-3.5-turbo-16k" "gpt-4" "gpt-4-1106-preview") "https://api.openai.com/v1/chat/completions")
 
The three question-mark figures are as follows: \300\301\302 and \207.

Here is the gptel setup in my init.el: 

(use-package gptel
  :if (is-mbp-p)
  :config 
   (setq-default gtpel-backend 
		(gptel-make-openai                    ;Not a typo, same API as OpenAI
		 "llama-cpp"                          ;Any name
		 :stream t                            ;Stream responses
		 :protocol "http"
		 :host "127.0.0.1:8080"               ;Llama.cpp server location, typically localhost:8080 for Llamafile
		 :models '("test"))))           ;Any names, doesn't matter for Llama

Thanks for answering!

**Reply by karthink** [reply]
Ah, I see the problem.  This is a bug, thanks for the report, I'll fix it soon.

**Reply by eastbowl** [reply]
My first ever bug report!  Hope it's not something big to fix.  Thank you.

**Comment by karthink** [comment]
Created issue in #170
